pallets/asset-manager/src/benchmarks.rs

```diff
@@ -45,6 +45,19 @@ benchmarks! {
 	verify {
 		assert_eq!(Pallet::<T>::asset_id_units_per_second(asset_id), Some(1));
 	}
+
+	change_existing_asset_type {
+		// does not really matter what we register
+		let asset_type = T::AssetType::default();
+		let metadata = T::AssetRegistrarMetadata::default();
+		let amount = 1u32.into();
+		let asset_id: T::AssetId = asset_type.clone().into();
+		Pallet::<T>::register_asset(RawOrigin::Root.into(), asset_type.clone(), metadata, amount, true)?;
+
+	}: _(RawOrigin::Root, asset_id, asset_type.clone())
+	verify {
+		assert_eq!(Pallet::<T>::asset_id_type(asset_id), Some(asset_type));
```

Comment:
This looks like it's (1) changing from one asset type to the same asset type, and (2) expecting the implementation of `change_existing_asset_type()` to not short-circuit this case.

This works for now because that is actually how `change_existing_asset_type()` is implemented, but it's probably more future-proof to write this test so that it actually changes the type (that way a future "optimization" or similar won't cause the benchmark to become ineffective).

Reply:
I wanted to write this benchmark like you described, but I have two associated types (assetId, assetType) and in order to be generic (over the mock runtime and over the proper runtime where this is included) I can only generate the default one. Still I think the benchmarking is correct

Reply:
One option is to require that AssetType: From<MultiLocation> in the config, and in that case I can generate several. But to be honest the pallet is so simple that I did not even bother requiring this. Let me know if this is something we think we should add

Reply:
Agreed, I don't think my concern is worth a lot of effort...